File: client/ios/UseCase/Auth/Session/VerifySecondFactorUseCase.swift
public struct VerifySecondFactorUseCase {
    private let gateway: AuthSessionGateway

    public init(gateway: AuthSessionGateway) {
        self.gateway = gateway
    }

    public func callAsFunction(request: VerifySecondFactorRequest) async throws -> AuthSessionPayload {
        try await gateway.verifySecondFactor(request: request)
    }
}

File: client/android/UseCase/Auth/Session/VerifySecondFactorUseCase.kt
package app.mobiling.client.usecase.auth.session

import app.mobiling.client.contract.auth.session.AuthSessionPayload
import app.mobiling.client.contract.auth.session.VerifySecondFactorRequest
import app.mobiling.client.data.auth.session.AuthSessionGateway

class VerifySecondFactorUseCase(
    private val gateway: AuthSessionGateway,
) {
    suspend operator fun invoke(request: VerifySecondFactorRequest): AuthSessionPayload =
        gateway.verifySecondFactor(request)
}
